### PR TITLE
Initial auth capabilities

### PR DESCRIPTION
**Pull request by carlesarnal**
The quarkus.http.auth.permission.$POLICY_NAME.enabled property did the trick. We now have the capability of enabling/disabling authentication AND authorization entirely with ENV_VARS. No extra docker images needed, no need for additional systems.

This also includes support for the client credentials flow.

### CODE REVIEW

**Comment by carlesarnal** [comment]
@Apicurio/developers this is now ready for review.

File: app/src/main/resources/application.properties [hunk 1]
```diff
@@ -78,3 +78,41 @@ mp.openapi.servers=/api
...
+quarkus.http.auth.permission.admin.methods=PUT,POST,DELETE,PATCH
+
+quarkus.http.auth.permission.permit1.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.permit1.paths=/health,/openapi,/metrics/*
+quarkus.http.auth.permission.permit1.policy=permit
+quarkus.http.auth.permission.permit1.methods=GET
+
+quarkus.http.auth.permission.authenticated.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.authenticated.paths=/api/*
+quarkus.http.auth.permission.authenticated.policy=authenticated
```
Comment (EricWittmann) [inline app/src/main/resources/application.properties:117]: Can you iterate on this to match the description in the security ADR?  The roles defined there are a bit different than what we have here.  Also you'll need to make sure we're protecting both compatibility APIs as well (ccompat and ibmcompat).

**Comment by EricWittmann** [comment]
There seems to be a file called `CompatibleClient.java` in this PR that is empty (0 bytes).

File: storage/kafka/src/test/resources/cluster-test.properties [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+initializer=io.apicurio.registry.storage.util.KafkaServiceInitializer
+node1.props=-Dquarkus.profile=dev -Dbootstrap.servers=${bootstrap.servers} -Dquarkus.oidc.tenant-enabled=false
+node2.props=-Dquarkus.profile=dev -Dbootstrap.servers=${bootstrap.servers} -Dquarkus.http.port=8081 -Dquarkus.oidc.tenant-enabled=false
+end.1=Cluster ID:
+end.2=Resetting offset for partition storage-topic-0 to offset
```
Comment (EricWittmann) [inline storage/kafka/src/test/resources/cluster-test.properties:6]: This file should probably be deleted - the *kafka* variant has been removed.

File: app/src/main/resources/application.properties (hunk 1 above)
Comment (carlesarnal) [reply app/src/main/resources/application.properties]: I have adapted the permissions and tested the compatibility APIs.

Comment (EricWittmann) [reply app/src/main/resources/application.properties]: I think there should be 3 roles, no? 

* Read-Only
* Modify allowed (standard access)
* Admin (same as above + can modify global rules)

I'm not seeing that here yet.

Comment (carlesarnal) [reply app/src/main/resources/application.properties]: I think that you have reviewed an outdated version of the file. I've added the three requires roles.

Comment (EricWittmann) [reply app/src/main/resources/application.properties]: You were right, sorry about that.  I was definitely looking at the old stuff.  My bad.

**Comment by EricWittmann** [comment]
@carlesarnal Can you take a look at the integration test failure?  I haven't looked yet, but it's a failing check on the PR.

**Comment by carlesarnal** [comment]
> @carlesarnal Can you take a look at the integration test failure? I haven't looked yet, but it's a failing check on the PR.

Yes, sorry. It was a UI test failure. I've added the logout button but completely forgot to hide it when the auth is disabled. Added that check.

**Comment by famartinrh** [comment]
I started to work on adding some basic tests for the auth functionalities, for now I only focused on verifying authentication and authorization using the registry rest client. it's not finished and I may need some help... I'm leaving it here so it's not lost https://github.com/famartinrh/apicurio-registry/tree/auth-add-tests

Btw we still have to evaluate a complete plan for testing authentication and authorization...

**Comment by carlesarnal** [comment]
> I started to work on adding some basic tests for the auth functionalities, for now I only focused on verifying authentication and authorization using the registry rest client. it's not finished and I may need some help... I'm leaving it here so it's not lost https://github.com/famartinrh/apicurio-registry/tree/auth-add-tests
> 
> Btw we still have to evaluate a complete plan for testing authentication and authorization...

Maybe is a good idea if you open a draft PR so we can start discussing [this](https://github.com/Apicurio/apicurio-registry/issues/910).

**Comment by carlesarnal** [comment]
@EricWittmann unless you have any objections, I think this is good to merge now.

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**